### PR TITLE
Add Meson build

### PR DESCRIPTION
**Pull request by ret2libc**

### CODE REVIEW

File: meson.build [hunk 1]
```diff
@@ -0,0 +1,40 @@
+project('spp', 'c')
+
+platform_inc = ['.']
+c_args = []
+
+spp_files = [
+  'spp.c'
+]
+
+if get_option('use_r_util') and get_option('r_util_includes') != ''
```
Comment (radare) [inline meson.build]: im not sure about the point of this option, r_api.c is required to build for spp.c and it's definively confusing to have a subproject to have an option for a superproject.

Comment (ret2libc) [reply meson.build]: r_api is not required. Unfortunately SPP is a bit problematic. If you compile it standalone, r_api.c is used to define r_strbuf_*, r_sys_* and other used functions.

But when used inside radare2, r_strbuf_* and r_sys* functions are provided by r_util.

**Comment by radare** [comment]
Yep i know, this was kind of a hack to get this into r2 without too much hassle or dupplicated code. maybe the better solution would be to rename this r_api code (which is probably outdated compared to the current state of the implementation). all the code using r_* apis inside spp is not public, so we dont need to expose or make this reusable from outside imho, so adding this option just complicates the build system for just 3 functions that process strings. i would go for making them more specific to spp. what do you think?

**Comment by ret2libc** [comment]
I thought a bit about this problem, but I couldn't really come up with a good solution. The ideal solution IMHO would be to have a r_util library which can be self-compiled without spp/sdb/etc. and that that can be used by spp as well as other r_* libraries.

I'm not sure we are talking about the same thing though. Those options I added are not to make r_* functions in spp accessible from outside. They are a way for r2 to provide its own versions of r_* functions, instead of using the (old) embedded one in spp.

Are you suggesting that we just throw away the r_* functions in spp and we write simpler code there, that doesn't require them?

**Comment by radare** [comment]
spp only uses those functions to optimize the “strcat” with a stringbuffer style api. i didnt wanted to redo the same thing i did in r2, so i opted for copypasting that into r_api.c

imho i would go for making those functions spp-specific, so, not public, they are not that big, and we dont need to expose them, its just that we want a way to concat strings in an optimal way, to finally get a char*

**Comment by ret2libc** [comment]
There are also r_sys functions and others, but I’ll look better at the code and see if we can remove them

**Comment by ret2libc** [comment]
So what do you think of this? I just always use the local definitions of s_strbuf_* and s_sys_*.
This, however, means that when spp will be linked with radare2's code, there would be some "duplicated functions" that do the same stuff.

**Comment by radare** [comment]
Thats exactly what i said. Just rename them and make them static and non gloably visible. And probably make them simpler and more optimized for the only usecase we have

**Pull request merged**